src/components/header/header-cell.component.ts

```diff
@@ -141,11 +153,31 @@ export class DataTableHeaderCellComponent {
   _sorts: any[];
   selectFn = this.select.emit.bind(this.select);
 
+  private headerTemplateContext: HeaderTemplateContext;
+
+  private initialized: boolean;
+
   @HostListener('contextmenu', ['$event'])
   onContextmenu($event: MouseEvent): void {
     this.columnContextmenu.emit({ event: $event, column: this.column });
   }
 
+  ngOnInit() {
+    this.updateHeaderTemplateContext();
+    this.initialized = true;
```

Comment:
I think when I originally worked on this, ngOnChanges was called more than once during the Component startup lifecycle.
However this should be reviewed.